lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -103,6 +103,21 @@ LogicalResult NnModuleOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// PrimListConstructOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verify(PrimListConstructOp op) {
+  auto resultType = op.getResult().getType();
+  auto resultElementType = resultType.dyn_cast<ListType>().getContainedType();
+  auto matchResultElementType = [&](Type type) {
+    return type.classof(resultElementType);
```

Comment:
"classof" does not do virtual dispatch, so this is really a trivially true call (it calls [this function](https://github.com/llvm/llvm-project/blob/4d11ba38b47de1da1cee156a8bf8b5d3447326b9/mlir/include/mlir/IR/Types.h#L102)).

I think what you want is `type.getTypeID() == resultElementType.getTypeID()`.